src/Libraries/Microsoft.Extensions.Telemetry.Testing/Metering/MetricCollector.cs

```diff
@@ -130,18 +130,18 @@ public void Dispose()
             }
 
             _disposed = true;
+        }
 
-            _meterListener.Dispose();
-            _measurements.Clear();
-
-            // wake up anybody still waiting and inform them of the bad news: their horse is dead...
-            foreach (var w in _waiters)
-            {
-                _ = w.TaskSource.TrySetException(MakeObjectDisposedException());
-            }
+        _meterListener.Dispose();
+        _measurements.Clear();
 
-            _waiters.Clear();
+        // wake up anybody still waiting and inform them of the bad news: their horse is dead...
```

Comment:
could you please add a comment in all places we complete the waiter.TaskSource telling need to be called outside the lock.